go/logger/global.go

```diff
@@ -11,8 +14,30 @@ import (
 var globalLock sync.Mutex
 var stderrIsTerminal = isatty.IsTerminal(os.Stderr.Fd())
 var currentLogFileWriter *LogFileWriter
+var stdErrLoggingShutdown chan<- struct{}
+
+func bufferLogs(writer io.Writer) (io.Writer, chan<- struct{}) {
+	buf := bufio.NewWriter(writer)
+	shutdown := make(chan struct{})
+	go func() {
```

Comment:
Add a `defer buf.Flush()` here.

Reply:
if we put the `defer Flush()` in the goroutine here, it will never execute. See https://github.com/golang/go/issues/14542

Reply:
or https://play.golang.org/p/o31M7ElNK0Q which is how I convinced myself I was reading that issue correctly.

Reply:
TIL. I think we've been working under the opposite assumption to date.